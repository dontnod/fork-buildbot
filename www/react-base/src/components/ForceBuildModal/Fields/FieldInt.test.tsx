/*
  This file is part of Buildbot.  Buildbot is free software: you can
  redistribute it and/or modify it under the terms of the GNU General Public
  License as published by the Free Software Foundation, version 2.

  This program is distributed in the hope that it will be useful, but WITHOUT
  ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
  FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
  details.

  You should have received a copy of the GNU General Public License along with
  this program; if not, write to the Free Software Foundation, Inc., 51
  Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.

  Copyright Buildbot Team Members
*/

import {describe, expect, it} from "vitest";
import renderer from 'react-test-renderer';
import { FieldInt } from "./FieldInt";
import { ForceSchedulerFieldInt } from 'buildbot-data-js';
import { ForceBuildModalFieldsState } from '../ForceBuildModalFieldsState';

function assertRenderToSnapshot(defaultValue: number, stateValue?: number, updateValue?: number) {
  const field: ForceSchedulerFieldInt = {
    name: 'dummy',
    fullName: 'fullDummy',
    label: 'dummyLabel',
    tablabel: 'dummyTabLabel',
    type: 'int',
    default: defaultValue,
    multiple: false,
    regex: null,
    hide: false,
    maxsize: null,
    autopopulate: null,
    size: 0,
  }
  const state = new ForceBuildModalFieldsState();
  state.createNewField(field.fullName, field.default);
  if (stateValue !== undefined) {
    state.setValue(field.fullName, stateValue);
  }

  const component = renderer.create(
    <FieldInt field={field} fieldsState={state} />
  );
  expect(component.toJSON()).toMatchSnapshot();

  if (updateValue !== undefined) {
    const expectedState = updateValue;
    renderer.act(() => {
      const elements = component.root.findAllByProps({'data-bb-test-id': `force-field-${field.fullName}`}, {deep: true});
      expect(elements.length).toBe(1);
      const input = elements[0];
      input.props.onChange({target: {value: expectedState}});
    });
    expect(state.getValue(field.fullName)).toBe(expectedState);
  }
}

describe('ForceFieldInt component', function () {
  it('render default value', () => {
    assertRenderToSnapshot(0);
    assertRenderToSnapshot(-0);
    assertRenderToSnapshot(150);
    assertRenderToSnapshot(-150);
  });

  it('render non-default value', () => {
    assertRenderToSnapshot(0, -0);
    assertRenderToSnapshot(0, 150);
    assertRenderToSnapshot(0, -150);
  });

  it('change state on click', () => {
    assertRenderToSnapshot(0, -150, 350);
  });
});
