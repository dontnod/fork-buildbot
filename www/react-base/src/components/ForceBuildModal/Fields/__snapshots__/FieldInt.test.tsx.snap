// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ForceFieldInt component > change state on click 1`] = `
<div>
  <div
    className="form-group"
  >
    <div
      popover-is-open="field.haserrors"
      popover-title="{{field.label}}"
      popover-trigger="none"
      uib-popover="{{field.errors}}"
    />
    <label
      className="control-label col-sm-10"
      htmlFor="fullDummy"
    >
      dummyLabel
    </label>
    <div
      className="col-sm-10"
    >
      <input
        className="form-control"
        data-bb-test-id="force-field-fullDummy"
        onChange={[Function]}
        type="number"
        value={-150}
      />
    </div>
  </div>
</div>
`;

exports[`ForceFieldInt component > render default value 1`] = `
<div>
  <div
    className="form-group"
  >
    <div
      popover-is-open="field.haserrors"
      popover-title="{{field.label}}"
      popover-trigger="none"
      uib-popover="{{field.errors}}"
    />
    <label
      className="control-label col-sm-10"
      htmlFor="fullDummy"
    >
      dummyLabel
    </label>
    <div
      className="col-sm-10"
    >
      <input
        className="form-control"
        data-bb-test-id="force-field-fullDummy"
        onChange={[Function]}
        type="number"
        value={0}
      />
    </div>
  </div>
</div>
`;

exports[`ForceFieldInt component > render default value 2`] = `
<div>
  <div
    className="form-group"
  >
    <div
      popover-is-open="field.haserrors"
      popover-title="{{field.label}}"
      popover-trigger="none"
      uib-popover="{{field.errors}}"
    />
    <label
      className="control-label col-sm-10"
      htmlFor="fullDummy"
    >
      dummyLabel
    </label>
    <div
      className="col-sm-10"
    >
      <input
        className="form-control"
        data-bb-test-id="force-field-fullDummy"
        onChange={[Function]}
        type="number"
        value={-0}
      />
    </div>
  </div>
</div>
`;

exports[`ForceFieldInt component > render default value 3`] = `
<div>
  <div
    className="form-group"
  >
    <div
      popover-is-open="field.haserrors"
      popover-title="{{field.label}}"
      popover-trigger="none"
      uib-popover="{{field.errors}}"
    />
    <label
      className="control-label col-sm-10"
      htmlFor="fullDummy"
    >
      dummyLabel
    </label>
    <div
      className="col-sm-10"
    >
      <input
        className="form-control"
        data-bb-test-id="force-field-fullDummy"
        onChange={[Function]}
        type="number"
        value={150}
      />
    </div>
  </div>
</div>
`;

exports[`ForceFieldInt component > render default value 4`] = `
<div>
  <div
    className="form-group"
  >
    <div
      popover-is-open="field.haserrors"
      popover-title="{{field.label}}"
      popover-trigger="none"
      uib-popover="{{field.errors}}"
    />
    <label
      className="control-label col-sm-10"
      htmlFor="fullDummy"
    >
      dummyLabel
    </label>
    <div
      className="col-sm-10"
    >
      <input
        className="form-control"
        data-bb-test-id="force-field-fullDummy"
        onChange={[Function]}
        type="number"
        value={-150}
      />
    </div>
  </div>
</div>
`;

exports[`ForceFieldInt component > render non-default value 1`] = `
<div>
  <div
    className="form-group"
  >
    <div
      popover-is-open="field.haserrors"
      popover-title="{{field.label}}"
      popover-trigger="none"
      uib-popover="{{field.errors}}"
    />
    <label
      className="control-label col-sm-10"
      htmlFor="fullDummy"
    >
      dummyLabel
    </label>
    <div
      className="col-sm-10"
    >
      <input
        className="form-control"
        data-bb-test-id="force-field-fullDummy"
        onChange={[Function]}
        type="number"
        value={-0}
      />
    </div>
  </div>
</div>
`;

exports[`ForceFieldInt component > render non-default value 2`] = `
<div>
  <div
    className="form-group"
  >
    <div
      popover-is-open="field.haserrors"
      popover-title="{{field.label}}"
      popover-trigger="none"
      uib-popover="{{field.errors}}"
    />
    <label
      className="control-label col-sm-10"
      htmlFor="fullDummy"
    >
      dummyLabel
    </label>
    <div
      className="col-sm-10"
    >
      <input
        className="form-control"
        data-bb-test-id="force-field-fullDummy"
        onChange={[Function]}
        type="number"
        value={150}
      />
    </div>
  </div>
</div>
`;

exports[`ForceFieldInt component > render non-default value 3`] = `
<div>
  <div
    className="form-group"
  >
    <div
      popover-is-open="field.haserrors"
      popover-title="{{field.label}}"
      popover-trigger="none"
      uib-popover="{{field.errors}}"
    />
    <label
      className="control-label col-sm-10"
      htmlFor="fullDummy"
    >
      dummyLabel
    </label>
    <div
      className="col-sm-10"
    >
      <input
        className="form-control"
        data-bb-test-id="force-field-fullDummy"
        onChange={[Function]}
        type="number"
        value={-150}
      />
    </div>
  </div>
</div>
`;
